tests/utils/set_params.cpp

```diff
@@ -1186,8 +1186,8 @@ void setStaggeredMultigridParam(QudaMultigridParam &mg_param)
 
   // set file i/o parameters
   for (int i = 0; i < mg_param.n_level; i++) {
-    strcpy(mg_param.vec_infile[i], mg_vec_infile[i]);
-    strcpy(mg_param.vec_outfile[i], mg_vec_outfile[i]);
+    strcpy(mg_param.vec_infile[i], mg_vec_infile[i].c_str());
+    strcpy(mg_param.vec_outfile[i], mg_vec_outfile[i].c_str());
```

Comment:
Add bounds check here and use `size()` on the two lines below

Reply:
See the latest changes.